From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Fri, 27 Mar 2020 10:18:41 -0700
Subject: ANDROID: GKI: Add CONFIG_RPMSG to gki_defconfig

RPMSG is used on various SoC platforms downstream. Enable the subsystem
in the gki_defconfig. This also enables RPMSG_CHAR.

Bug: 146449535
Change-Id: Ib4a6e5a438681a9835398cdb8917045191e1fbe6
Signed-off-by: Alistair Delva <adelva@google.com>
[joneslee: Removed defconfig elements]
Signed-off-by: Lee Jones <lee.jones@linaro.org>
---
 init/Kconfig.gki | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -137,6 +137,17 @@ config GKI_HIDDEN_SOC_BUS_CONFIGS
 	    If you are not building a kernel to be used for a variety of SoCs and
 	    out-of-tree drivers, say N here.
 
+config GKI_HIDDEN_RPMSG_CONFIGS
+	bool "Hidden RPMSG configuration needed for GKI"
+	select RPMSG
+	help
+	  Dummy config option used to enable the hidden RPMSG config.
+	  This configuration is usually only selected by another config
+	  option or a combination of them.
+
+	  If you are not building a kernel to be used for a variety of
+	  out-of-kernel build RPMSG drivers, say N here.
+
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
 # b) building drivers that are not modularizable
@@ -159,6 +170,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_VIRTUAL_CONFIGS
 	select GKI_HIDDEN_USB_CONFIGS
 	select GKI_HIDDEN_SOC_BUS_CONFIGS
+	select GKI_HIDDEN_RPMSG_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
